kube3d: 3.0.2 -> 3.1.0 -> 3.1.1 -> 3.1.2 -> 3.1.3 -> 3.1.4 -> 3.1.5

06kellyjac opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change

Bump kube3d/k3d to ~`3.1.0`~ ~`3.1.1`~ ~`3.1.2`~ ~`3.1.3`~ ~`3.1.4`~ `3.1.5`

###### Things done

Updated version, k3s version, and src sha256
I matched the k3s version to the one output by the GH release binary, hopefully that's correct.

@kuznero @jlesquembre @ngerstle 

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS (x86_64)
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

06kellyjac committed:
kube3d: 3.0.2 -> 3.1.0

Also bumped k3sVersion

06kellyjac committed:
kube3d: 3.1.0 -> 3.1.1

06kellyjac committed:
kube3d: 3.1.1 -> 3.1.2

06kellyjac committed:
kube3d: 3.1.2 -> 3.1.3

06kellyjac committed:
kube3d: 3.1.3 -> 3.1.4

06kellyjac commented:
Friendly bump @kuznero @jlesquembre @ngerstle

06kellyjac committed:
kube3d: 3.1.4 -> 3.1.5

06kellyjac commented:
I've rebased on master & corrected my mistake with the k3sVersion having the `v` attached, also version bumped again

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/346

jlesquembre commented:
@06kellyjac Thanks for your PR, I didn't have time to test it, but since it's a simple version dump , it should be ok

06kellyjac commented:
Thanks @jlesquembre would you mind if I add myself as a maintainer too?

jlesquembre commented:
@06kellyjac go for it, better to have more maintainers :-)

06kellyjac committed:
kube3d: add jk as a maintainer

Add jk (myself) as a maintainer to help keeping kube3d working and
up-to-date.

06kellyjac commented:
Added :slightly_smiling_face:

jlesquembre commented on pkgs/applications/networking/cluster/kube3d/default.nix:
```diff
@@ -2,16 +2,16 @@
-  version = "3.0.2";
-  k3sVersion = "1.18.6-k3s1";
+  version = "3.1.5";
+  k3sVersion = "1.18.9-k3s1";
```
Latest k3s version is 1.18.10, should we use that one?

06kellyjac replied:
I've just been copying whatever value is in the released k3d binary, the image can be overriden with the `--image` flag when creating the cluster

jlesquembre replied:
Fair enough, makes sense to align with the upstream version

jlesquembre reviewed (commented):
I notice that k3d added completions for fish, can you add it too?
```
    for shell in bash zsh fish; do
```
Other than that, all looks good, tested it locally

06kellyjac commented:
Sure, I'll pop that in tomorrow morning

06kellyjac committed:
kube3d: add completion for fish

fish completion was added in 3.0.2
https://github.com/rancher/k3d/releases/tag/v3.0.2

06kellyjac commented:
![image](https://user-images.githubusercontent.com/9866621/97155567-7662e980-176d-11eb-9beb-48348aff32a1.png)

I added it to the start of this PR as the feature was added in `3.0.2`

jlesquembre commented on pkgs/applications/networking/cluster/kube3d/default.nix:
```diff
@@ -22,11 +22,13 @@ buildGoModule rec {
 
   nativeBuildInputs = [ installShellFiles ];
+
+  # TODO: Move to enhanced installShellCompletion when in master: PR #83630
```
Isn't #83630 already merged?

06kellyjac replied:
It's not in master yet
![image](https://user-images.githubusercontent.com/9866621/97157780-76181d80-1770-11eb-91dd-4865c713e065.png)

jlesquembre replied:
right, my bad, it's only on staging

jlesquembre commented:
/marvin opt-in
/status needs_merger

jlesquembre commented:
/marvin opt-in

jlesquembre commented:
@06kellyjac I was trying to use marvin: https://github.com/timokau/marvin-mk2/blob/master/USAGE.md
but looks like the PR author needs to do it, can you try?
I tested it and looks good to me, but I don't have merge rights on nixpkgs.

06kellyjac commented:
/marvin opt-in

06kellyjac commented:
/marvin needs_merger

06kellyjac commented:
/status needs_merger

jlesquembre commented:
/status needs_merger

06kellyjac commented:
Theres also a discourse thread for PRs that are reviewed but need a merger. I'd grab the link but I'm on mobile atm.

timokau reviewed (approved):
Thanks!

timokau commented:
> Theres also a discourse thread for PRs that are reviewed but need a merger. I'd grab the link but I'm on mobile atm.

Marvin is intended to be a better replacement for that. It
- helps you find the initial review as well
- is higher signal, lower noise (this is mostly from a reviewer/"commiter" perspective since marvin can actually verify that there *was* a 3rd-party review)
- distributes the work better by assigning reviewers / "commiters".

Its still experimental though and will likely remain that way for quite a while.

06kellyjac commented:
Does marvin have a dashboard too or is it just good at only assigning tags if they're valid & replying well to incorrect commands?

timokau commented:
I'm not sure what you mean. There is no dashboard. What information would you want displayed? If you want to find PRs to review, you can use this filter: https://github.com/NixOS/nixpkgs/labels/needs_reviewer

It usually does not reply to misformatted commands (it should eventually do that, but doesn't for now). The best source of "end user" documentation is probably [the bot's "profile"](https://github.com/apps/marvin-mk2) right now.

06kellyjac commented:
Yeah I was just wondering if it was adding good tags for filtering or if there was something else it was providing for nixpkgs pr observability that i didnt know about

I dont mean misformatted. I mean it told me when i couldn't change to awaiting merger and that it wanted the reviewer to post it. Which was helpful

06kellyjac commented:
If i put /m*rvin opt-in in the original PR description will it work or does it have to be a comment?
